iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ThreeWindingsTransformerImpl.java

```diff
@@ -166,18 +115,18 @@ public Leg1Impl setB(double b) {
             return this;
         }
 
-        @Override
-        protected String getLegAttribute() {
-            return "leg1";
+        public double getRatedU() {
+            return ratedU;
         }
-    }
-
-    private abstract static class AbstractLeg2or3<T extends AbstractLeg2or3<T>> extends AbstractLegBase<T> implements RatioTapChangerParent {
-
-        private RatioTapChangerImpl ratioTapChanger;
 
-        AbstractLeg2or3(double r, double x, double ratedU) {
-            super(r, x, ratedU);
+        public Leg setRatedU(double ratedU) {
+            if (Double.isNaN(ratedU)) {
+                throw new ValidationException(this, "rated U is invalid");
+            }
```

Comment:
Could you refactor the `checkRatedU1` and `checkRatedU2`  in `ValidationUtil` as such:
```java
static void checkRatedU(Validable validable, double ratedU, String num) {
        if (Double.isNaN(ratedU)) {
            throw new ValidationException(validable, "rated U" + num + " is invalid");
        }
    }

static void checkRatedU1(Validable validable, double ratedU1) {
    checkRatedU(validable, ratedU1, "1");
}

static void checkRatedU2(Validable validable, double ratedU2) {
  checkRatedU(validable, ratedU2, "2");
}
```
This way, you could use `ValidationUtil.checkRatedU(this, ratedU, "");` here

Reply:
Done